runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -187,23 +170,28 @@ public void onExperienceChanged(ExperienceChanged event)
 
 		// Get course
 		Courses course = Courses.getCourse(client.getLocalPlayer().getWorldLocation().getRegionID());
-		if (course == null
-			|| (course.getCourseEndWorldPoints().length == 0
-			? Math.abs(course.getLastObstacleXp() - skillGained) > 1
-			: Arrays.stream(course.getCourseEndWorldPoints()).noneMatch(wp -> wp.equals(client.getLocalPlayer().getWorldLocation()))))
-		{
+		if (course == null)
 			return;
+
+		// Get new agility session
+		if ( !(session != null && session.getCourse() == course))
```

Comment:
This seems convoluted... Why not `session == null || session.getCourse() != course`?

Reply:
resolved in latest commit